docker imagefailed to build on macOS Catalina

**Please describe the bug:**  
`docker-compose build` is failing, leaving me with an image in incorrect state.

**Please describe steps to reproduce:**
(Following the `README.md`)
1. Run `docker-compose build
2. Wait until the CMake build process is at around 79% 
3. See build failures happen

**What is the expected behavior? Please describe:**  
I would expect a working docker-compose built image.

**Affected Component(s):**  
`Docker Image`

**Log excerpt:**  

```
[ 79%] Building CXX object Source/CMakeFiles/CMakeLib.dir/cmProjectCommand.cxx.o
[ 79%] Building CXX object Source/CMakeFiles/CMakeLib.dir/cmRemoveCommand.cxx.o
[ 79%] Building CXX object Source/CMakeFiles/CMakeLib.dir/cmPathLabel.cxx.o
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:258: Source/CMakeFiles/CMakeLib.dir/cmBinUtilsLinuxELFLinker.cxx.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 79%] Building CXX object Source/CMakeFiles/CMakeLib.dir/cmRemoveDefinitionsCommand.cxx.o
[ 79%] Building CXX object Source/CMakeFiles/CMakeLib.dir/cmReturnCommand.cxx.o
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:427: Source/CMakeFiles/CMakeLib.dir/cmCommonTargetGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:388: Source/CMakeFiles/CMakeLib.dir/cmCheckCustomOutputs.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:375: Source/CMakeFiles/CMakeLib.dir/cmCacheManager.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:349: Source/CMakeFiles/CMakeLib.dir/cmBinUtilsWindowsPELinker.cxx.o] Error 1
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:453: Source/CMakeFiles/CMakeLib.dir/cmComputeLinkDepends.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:596: Source/CMakeFiles/CMakeLib.dir/cmDependsC.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:414: Source/CMakeFiles/CMakeLib.dir/cmCommandArgumentParserHelper.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:297: Source/CMakeFiles/CMakeLib.dir/cmBinUtilsMacOSMachOLinker.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:479: Source/CMakeFiles/CMakeLib.dir/cmComputeTargetDepends.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:544: Source/CMakeFiles/CMakeLib.dir/cmCustomCommandGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
[ 79%] Building CXX object Source/CMakeFiles/CMakeLib.dir/cmSearchPath.cxx.o
g++: fatal error: Killed signal terminated program cc1plus
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:466: Source/CMakeFiles/CMakeLib.dir/cmComputeLinkInformation.cxx.o] Error 1
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:570: Source/CMakeFiles/CMakeLib.dir/cmDefinitions.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:895: Source/CMakeFiles/CMakeLib.dir/cmFileAPI.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:947: Source/CMakeFiles/CMakeLib.dir/cmFileCopier.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:765: Source/CMakeFiles/CMakeLib.dir/cmExportInstallAndroidMKGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:830: Source/CMakeFiles/CMakeLib.dir/cmExtraCodeBlocksGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:440: Source/CMakeFiles/CMakeLib.dir/cmComputeComponentGraph.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:882: Source/CMakeFiles/CMakeLib.dir/cmExtraSublimeTextGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:635: Source/CMakeFiles/CMakeLib.dir/cmDependsJavaParserHelper.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:921: Source/CMakeFiles/CMakeLib.dir/cmFileAPICodemodel.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:609: Source/CMakeFiles/CMakeLib.dir/cmDependsFortran.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:791: Source/CMakeFiles/CMakeLib.dir/cmExportTryCompileFileGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:908: Source/CMakeFiles/CMakeLib.dir/cmFileAPICache.cxx.o] Error 1
compilation terminated.
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:700: Source/CMakeFiles/CMakeLib.dir/cmELF.cxx.o] Error 1
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:752: Source/CMakeFiles/CMakeLib.dir/cmExportFileGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:583: Source/CMakeFiles/CMakeLib.dir/cmDepends.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:518: Source/CMakeFiles/CMakeLib.dir/cmCurl.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:726: Source/CMakeFiles/CMakeLib.dir/cmExportBuildAndroidMKGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:869: Source/CMakeFiles/CMakeLib.dir/cmExtraKateGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:934: Source/CMakeFiles/CMakeLib.dir/cmFileAPICMakeFiles.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:804: Source/CMakeFiles/CMakeLib.dir/cmExportSet.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:856: Source/CMakeFiles/CMakeLib.dir/cmExtraEclipseCDT4Generator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1168: Source/CMakeFiles/CMakeLib.dir/cmGeneratorExpressionNode.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1363: Source/CMakeFiles/CMakeLib.dir/cmInstallSubdirectoryGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1142: Source/CMakeFiles/CMakeLib.dir/cmGeneratorExpressionEvaluator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:739: Source/CMakeFiles/CMakeLib.dir/cmExportBuildFileGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1584: Source/CMakeFiles/CMakeLib.dir/cmMakefileUtilityTargetGenerator.cxx.o] Error 1
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1298: Source/CMakeFiles/CMakeLib.dir/cmInstallGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1246: Source/CMakeFiles/CMakeLib.dir/cmGlobalGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1324: Source/CMakeFiles/CMakeLib.dir/cmInstalledFile.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1389: Source/CMakeFiles/CMakeLib.dir/cmInstallDirectoryGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:531: Source/CMakeFiles/CMakeLib.dir/cmCustomCommand.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1467: Source/CMakeFiles/CMakeLib.dir/cmListFileCache.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:778: Source/CMakeFiles/CMakeLib.dir/cmExportInstallFileGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1883: Source/CMakeFiles/CMakeLib.dir/cmSourceFileLocation.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1311: Source/CMakeFiles/CMakeLib.dir/cmInstallExportGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1935: Source/CMakeFiles/CMakeLib.dir/cmStateSnapshot.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1662: Source/CMakeFiles/CMakeLib.dir/cmOrderDirectories.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1792: Source/CMakeFiles/CMakeLib.dir/cmQtAutoGenInitializer.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1116: Source/CMakeFiles/CMakeLib.dir/cmGeneratorExpressionDAGChecker.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1012: Source/CMakeFiles/CMakeLib.dir/cmFilePathChecksum.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1909: Source/CMakeFiles/CMakeLib.dir/cmState.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1532: Source/CMakeFiles/CMakeLib.dir/cmMakefile.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1675: Source/CMakeFiles/CMakeLib.dir/cmPolicies.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1779: Source/CMakeFiles/CMakeLib.dir/cmQtAutoGenGlobalInitializer.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1129: Source/CMakeFiles/CMakeLib.dir/cmGeneratorExpressionEvaluationFile.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1259: Source/CMakeFiles/CMakeLib.dir/cmGlobalUnixMakefileGenerator3.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1558: Source/CMakeFiles/CMakeLib.dir/cmMakefileExecutableTargetGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:2273: Source/CMakeFiles/CMakeLib.dir/cmAddDependenciesCommand.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1402: Source/CMakeFiles/CMakeLib.dir/cmLDConfigLDConfigTool.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1571: Source/CMakeFiles/CMakeLib.dir/cmMakefileLibraryTargetGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1545: Source/CMakeFiles/CMakeLib.dir/cmMakefileTargetGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1285: Source/CMakeFiles/CMakeLib.dir/cmGraphVizWriter.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:2689: Source/CMakeFiles/CMakeLib.dir/cmFindPathCommand.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:2299: Source/CMakeFiles/CMakeLib.dir/cmAddLibraryCommand.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1207: Source/CMakeFiles/CMakeLib.dir/cmGeneratorTarget.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1454: Source/CMakeFiles/CMakeLib.dir/cmLinkLineDeviceComputer.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1623: Source/CMakeFiles/CMakeLib.dir/cmOSXBundleGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1805: Source/CMakeFiles/CMakeLib.dir/cmQtAutoMocUic.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:843: Source/CMakeFiles/CMakeLib.dir/cmExtraCodeLiteGenerator.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:2546: Source/CMakeFiles/CMakeLib.dir/cmExecuteProcessCommand.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:3001: Source/CMakeFiles/CMakeLib.dir/cmLinkDirectoriesCommand.cxx.o] Error 1
g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [Source/CMakeFiles/CMakeLib.dir/build.make:1233: Source/CMakeFiles/CMakeLib.dir/cmGlobalCommonGenerator.cxx.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2371: Source/CMakeFiles/CMakeLib.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
ERROR: Service 'hyrise' failed to build: The command '/bin/sh -c apt-get update && apt-get install -y wget libssl-dev && apt-get clean     && wget https://github.com/Kitware/CMake/releases/download/v3.16.0/cmake-3.16.0.tar.gz     && tar xf cmake-3.16.0.tar.gz     && cd cmake-3.16.0     && ./configure && make -j && make install' returned a non-zero code: 2
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -50,6 +50,6 @@ RUN apt-get update && apt-get install -y wget libssl-dev && apt-get clean \
     && wget https://github.com/Kitware/CMake/releases/download/v3.16.0/cmake-3.16.0.tar.gz \
     && tar xf cmake-3.16.0.tar.gz \
     && cd cmake-3.16.0 \
-    && ./configure && make -j && make install
+    && ./configure && make -j $(nproc) && make install
 
 ENV OPOSSUM_HEADLESS_SETUP=true

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -96,12 +96,12 @@ try {
 
         parallel clangDebug: {
           stage("clang-debug") {
-            sh "cd clang-debug && make all -j \$(( \$(cat /proc/cpuinfo | grep processor | wc -l) / 4))"
+            sh "cd clang-debug && make all -j \$(( \$(nproc) / 4))"
             sh "./clang-debug/hyriseTest clang-debug"
           }
         }, gccDebug: {
           stage("gcc-debug") {
-            sh "export CCACHE_BASEDIR=`pwd`; cd gcc-debug && ${disable_aslr} make all -j \$(( \$(cat /proc/cpuinfo | grep processor | wc -l) / 4)) && ../scripts/analyze_ccache_usage.py"
+            sh "export CCACHE_BASEDIR=`pwd`; cd gcc-debug && ${disable_aslr} make all -j \$(( \$(nproc) / 4)) && ../scripts/analyze_ccache_usage.py"
             sh "cd gcc-debug && ./hyriseTest"
           }
         }, lint: {
@@ -115,7 +115,7 @@ try {
         parallel clangRelease: {
           stage("clang-release") {
             if (env.BRANCH_NAME == 'master' || full_ci) {
-              sh "cd clang-release && make all -j \$(( \$(cat /proc/cpuinfo | grep processor | wc -l) / 6))"
+              sh "cd clang-release && make all -j \$(( \$(nproc) / 6))"
               sh "./clang-release/hyriseTest clang-release"
               sh "./clang-release/hyriseSystemTest clang-release"
               sh "./scripts/test/hyriseConsole_test.py clang-release"
@@ -160,7 +160,7 @@ try {
           stage("clang-debug-unity-odr") {
             if (env.BRANCH_NAME == 'master' || full_ci) {
               // Check if unity builds work even if everything is batched into a single compilation unit. This helps prevent ODR (one definition rule) issues.
-              sh "cd clang-debug-unity-odr && make all -j \$(( \$(cat /proc/cpuinfo | grep processor | wc -l) / 3))"
+              sh "cd clang-debug-unity-odr && make all -j \$(( \$(nproc) / 3))"
             } else {
               Utils.markStageSkippedForConditional("clangDebugUnityODR")
             }
@@ -169,7 +169,7 @@ try {
           stage("clang-debug:tidy") {
             if (env.BRANCH_NAME == 'master' || full_ci) {
               // We do not run tidy checks on the src/test folder, so there is no point in running the expensive clang-tidy for those files
-              sh "cd clang-debug-tidy && make hyrise_impl hyriseBenchmarkFileBased hyriseBenchmarkTPCH hyriseBenchmarkTPCDS hyriseBenchmarkJoinOrder hyriseConsole hyriseServer -k -j \$(( \$(cat /proc/cpuinfo | grep processor | wc -l) / 6))"
+              sh "cd clang-debug-tidy && make hyrise_impl hyriseBenchmarkFileBased hyriseBenchmarkTPCH hyriseBenchmarkTPCDS hyriseBenchmarkJoinOrder hyriseConsole hyriseServer -k -j \$(( \$(nproc) / 6))"
             } else {
               Utils.markStageSkippedForConditional("clangDebugTidy")
             }
@@ -178,7 +178,7 @@ try {
           stage("clang-debug:disable-precompile-headers") {
             if (env.BRANCH_NAME == 'master' || full_ci) {
               // Check if builds work even when precompile headers is turned off. Executing the binaries is unnecessary as the observed errors are missing includes.
-              sh "cd clang-debug-disable-precompile-headers && make hyriseTest hyriseBenchmarkFileBased hyriseBenchmarkTPCH hyriseBenchmarkTPCDS hyriseBenchmarkJoinOrder hyriseConsole hyriseServer -k -j \$(( \$(cat /proc/cpuinfo | grep processor | wc -l) / 6))"
+              sh "cd clang-debug-disable-precompile-headers && make hyriseTest hyriseBenchmarkFileBased hyriseBenchmarkTPCH hyriseBenchmarkTPCDS hyriseBenchmarkJoinOrder hyriseConsole hyriseServer -k -j \$(( \$(nproc) / 6))"
             } else {
               Utils.markStageSkippedForConditional("clangDebugDisablePrecompileHeaders")
             }
@@ -186,7 +186,7 @@ try {
         }, clangDebugAddrUBSanitizers: {
           stage("clang-debug:addr-ub-sanitizers") {
             if (env.BRANCH_NAME == 'master' || full_ci) {
-              sh "cd clang-debug-addr-ub-sanitizers && make hyriseTest hyriseSystemTest hyriseBenchmarkTPCH hyriseBenchmarkTPCC -j \$(( \$(cat /proc/cpuinfo | grep processor | wc -l) / 6))"
+              sh "cd clang-debug-addr-ub-sanitizers && make hyriseTest hyriseSystemTest hyriseBenchmarkTPCH hyriseBenchmarkTPCC -j \$(( \$(nproc) / 6))"
               sh "LSAN_OPTIONS=suppressions=resources/.lsan-ignore.txt ASAN_OPTIONS=suppressions=resources/.asan-ignore.txt ./clang-debug-addr-ub-sanitizers/hyriseTest clang-debug-addr-ub-sanitizers"
               sh "LSAN_OPTIONS=suppressions=resources/.lsan-ignore.txt ASAN_OPTIONS=suppressions=resources/.asan-ignore.txt ./clang-debug-addr-ub-sanitizers/hyriseSystemTest ${tests_excluded_in_sanitizer_builds} clang-debug-addr-ub-sanitizers"
               sh "LSAN_OPTIONS=suppressions=resources/.lsan-ignore.txt ASAN_OPTIONS=suppressions=resources/.asan-ignore.txt ./clang-debug-addr-ub-sanitizers/hyriseBenchmarkTPCH -s .01 --verify -r 1"
@@ -197,7 +197,7 @@ try {
         }, gccRelease: {
           if (env.BRANCH_NAME == 'master' || full_ci) {
             stage("gcc-release") {
-              sh "export CCACHE_BASEDIR=`pwd`; cd gcc-release && ${disable_aslr} make all -j \$(( \$(cat /proc/cpuinfo | grep processor | wc -l) / 6)) && ../scripts/analyze_ccache_usage.py"
+              sh "export CCACHE_BASEDIR=`pwd`; cd gcc-release && ${disable_aslr} make all -j \$(( \$(nproc) / 6)) && ../scripts/analyze_ccache_usage.py"
               sh "./gcc-release/hyriseTest gcc-release"
               sh "./gcc-release/hyriseSystemTest gcc-release"
               sh "./scripts/test/hyriseConsole_test.py gcc-release"
@@ -212,7 +212,7 @@ try {
         }, clangReleaseAddrUBSanitizers: {
           stage("clang-release:addr-ub-sanitizers") {
             if (env.BRANCH_NAME == 'master' || full_ci) {
-              sh "cd clang-release-addr-ub-sanitizers && make hyriseTest hyriseSystemTest hyriseBenchmarkTPCH hyriseBenchmarkTPCC -j \$(( \$(cat /proc/cpuinfo | grep processor | wc -l) / 6))"
+              sh "cd clang-release-addr-ub-sanitizers && make hyriseTest hyriseSystemTest hyriseBenchmarkTPCH hyriseBenchmarkTPCC -j \$(( \$(nproc) / 6))"
               sh "LSAN_OPTIONS=suppressions=resources/.lsan-ignore.txt ASAN_OPTIONS=suppressions=resources/.asan-ignore.txt ./clang-release-addr-ub-sanitizers/hyriseTest clang-release-addr-ub-sanitizers"
               sh "LSAN_OPTIONS=suppressions=resources/.lsan-ignore.txt ASAN_OPTIONS=suppressions=resources/.asan-ignore.txt ./clang-release-addr-ub-sanitizers/hyriseSystemTest ${tests_excluded_in_sanitizer_builds} clang-release-addr-ub-sanitizers"
               sh "LSAN_OPTIONS=suppressions=resources/.lsan-ignore.txt ASAN_OPTIONS=suppressions=resources/.asan-ignore.txt ./clang-release-addr-ub-sanitizers/hyriseBenchmarkTPCH -s .01 --verify -r 100 --scheduler --clients 10"
@@ -224,7 +224,7 @@ try {
         }, clangRelWithDebInfoThreadSanitizer: {
           stage("clang-relwithdebinfo:thread-sanitizer") {
             if (env.BRANCH_NAME == 'master' || full_ci) {
-              sh "cd clang-relwithdebinfo-thread-sanitizer && make hyriseTest hyriseSystemTest hyriseBenchmarkTPCH -j \$(( \$(cat /proc/cpuinfo | grep processor | wc -l) / 6))"
+              sh "cd clang-relwithdebinfo-thread-sanitizer && make hyriseTest hyriseSystemTest hyriseBenchmarkTPCH -j \$(( \$(nproc) / 6))"
               sh "TSAN_OPTIONS=\"history_size=7 suppressions=resources/.tsan-ignore.txt\" ./clang-relwithdebinfo-thread-sanitizer/hyriseTest clang-relwithdebinfo-thread-sanitizer"
               sh "TSAN_OPTIONS=\"history_size=7 suppressions=resources/.tsan-ignore.txt\" ./clang-relwithdebinfo-thread-sanitizer/hyriseSystemTest ${tests_excluded_in_sanitizer_builds} clang-relwithdebinfo-thread-sanitizer"
               sh "TSAN_OPTIONS=\"history_size=7 suppressions=resources/.tsan-ignore.txt\" ./clang-relwithdebinfo-thread-sanitizer/hyriseBenchmarkTPCH -s .01 --verify -r 100 --scheduler --clients 10"

